app/models/keyword.rb

```diff
@@ -4,8 +4,17 @@ class Keyword < ApplicationRecord
   include Discard::Model
 
   validates :name, presence: true, length: { maximum: 255 }
+  validates :user, presence: true, unless: :bulk_insert?
```

Comment:
Why do we need to disable this validation in case of bulk_insert? 🤔

Reply:
When we run, the rails validations are skipped anyways. Only database level constraints matter

Reply:
Short answer: in order to use the keyword model validation **before** insert_all `keywords.select(&:validate)`

If we base only on DB level validation, DB will trim any keyword longer than 255 (not to mention that the whole model validation is set for nothing). 
Though, when doing the model validation, for performance reason, I still need to bypass the user association validation (which is well managed by DB -- so it should not be a big deal).

Reply:
@malparty Got it, altough it can be good to checkout db level constraints for this case. https://blog.saeloun.com/2021/01/08/rails-6-check-constraints-database-migrations.html

```
   add_check_constraint :keywords, "char_length(name) <= 255", "char_length_name"
```

Reply:
@ankitkalia1195 wise solution. So to confirm, the idea is to not use keyword model validation at all and simply rescue DB exceptions to show eventual errors to the user, right? 
I'll update the code accordingnly. 
Meanwhile I set this PR back to `WIP`. Thanks!

Reply:
@malparty Yea, if we are using bulk insert, then we can skip the validations all together. But on our side we should be carefull of what validations and callbacks we are skipping

Reply:
@ankitkalia1195  thanks, very insightful :) 
I have updated the code accordingly:
- Rescue DB error, add the error (so that it can be displayed later)
- Add tests to ensure the transaction does not save any keyword if one of them could not be saved
- Remove "blank" validation and simply ignore any blank keyword (opportunity to discover `Array.filter_map` :) 

The PR is back to "in review" (no more WIP). 
Thanks :)